### PR TITLE
Help Center: Add long message validation error

### LINKED ISSUE #95584: Limit Odie message size
### Quick summary

Currently, users can send massive messages to Odie, the API does not allow that. Maybe we should prevent it from the front end too.

Image

### Steps to reproduce

Use help center
Send a massive message to Odie

### What you expected to happen

The message should be sent

### What actually happened

Message not sent

### Impact

One

### Available workarounds?

No and the platform is unusable

### If the above answer is "Yes...", outline the workaround.

_No response_

### Platform (Simple and/or Atomic)

_No response_

### Logs or notes

_No response_

### PR DESCRIPTION
**Pull request by renancarvalho**
Related to https://github.com/Automattic/wp-calypso/issues/95584
Fixes https://github.com/Automattic/wp-calypso/issues/95584

## Proposed Changes

* Prevent long messages (4096 chars) from being sent since the API doesn't accept it.

## Why are these changes being made?

* Prevent API errors and display to users that long messages are not allowed

## Testing Instructions

* Use live link and add flag `?flags=help-center-experience`
* Open a chat and type a really long message
* You should get a message explaining that the message is long

image

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: packages/odie-client/src/components/send-message-input/index.tsx [hunk 1]
```diff
@@ -22,9 +22,27 @@ export const OdieSendMessageButton = ( {
 	const { trackEvent, chatStatus, shouldUseHelpCenterExperience } = useOdieAssistantContext();
 	const sendMessage = useSendChatMessage();
 	const shouldBeDisabled = chatStatus === 'loading' || chatStatus === 'sending';
+	const [ isMessageSizeValid, setIsMessageSizeValid ] = useState( true );
+
+	const isMessageExceedingMaxLength = ( messageLength: number ) => {
+		// zendesk api validation
+		return messageLength <= 4096;
+	};
```
Comment (arcangelini) [inline packages/odie-client/src/components/send-message-input/index.tsx]: It doesn't look like you are using this in multiple places, and its pretty simple logic. Would you want to just put this on L41 as:

```
const isMessageLengthValid = messageLength <= 4096; // zendesk api validation
```

File: packages/odie-client/src/components/send-message-input/index.tsx [hunk 2]
```diff
@@ -22,9 +22,27 @@ export const OdieSendMessageButton = ( {
...
 
 	const sendMessageHandler = useCallback( async () => {
-		if ( inputRef.current?.value.trim() === '' || shouldBeDisabled ) {
+		const message = inputRef.current?.value.trim();
+		const messageLength = message?.length || 0;
+		const isMessageLengthValid = isMessageExceedingMaxLength( messageLength );
+
+		setIsMessageSizeValid( isMessageLengthValid );
+
+		if ( message === '' || shouldBeDisabled || ! isMessageLengthValid ) {
```
Comment (arcangelini) [inline packages/odie-client/src/components/send-message-input/index.tsx]:
```suggestion
		if ( message === '' || shouldBeDisabled || ( shouldUseHelpCenterExperience && ! isMessageLengthValid ) ) {
```

Do we want to only apply this to our flag for now?

**Review by arcangelini — APPROVED** [review]
Everything seems to work well for me. Just two suggestions. Nothing major.

File: packages/odie-client/src/components/send-message-input/index.tsx (hunk 1 above)
Comment (renancarvalho) [reply packages/odie-client/src/components/send-message-input/index.tsx]: Makes sense 👍

File: packages/odie-client/src/components/send-message-input/index.tsx (hunk 2 above)
Comment (renancarvalho) [reply packages/odie-client/src/components/send-message-input/index.tsx]: Addressed 👍

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/16982770

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @renancarvalho for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/16982770).

**Pull request merged**